Signal/src/call/CallAudioService.swift

```diff
@@ -216,14 +236,25 @@ struct AudioSource: Hashable {
                 try session.setPreferredInput(call.audioSource?.portDescription)
             }
 
-            if call.isSpeakerphoneEnabled {
+            if call.isSpeakerphoneEnabled || (call.hasLocalVideo && call.state != .connected) {
+                // We want consistent ringer-volume between speaker-phone and video chat.
+                // But because using VideoChat mode has noticeably higher output gain, we treat
+                // video chat like speakerphone mode until the call is connected.
+                Logger.verbose("\(TAG) enabling speakerphone overrideOutputAudioPort(.speaker)")
                 try session.overrideOutputAudioPort(.speaker)
             } else {
+                Logger.verbose("\(TAG) disabling spearkerphone overrideOutputAudioPort(.none) ")
                 try session.overrideOutputAudioPort(.none)
             }
         } catch {
             owsFail("\(TAG) failed setting audio source with error: \(error) isSpeakerPhoneEnabled: \(call.isSpeakerphoneEnabled)")
         }
+
+        if call.state == .connected, !call.isOnHold {
+            audioSession.isRTCAudioEnabled = true
+        } else {
+            audioSession.isRTCAudioEnabled = false
+        }
```

Comment:
Would this be simpler?
```
audioSession.isRTCAudioEnabled = call.state == .connected && !call.isOnHold

Reply:
Just a bit too dense for my arbitrary tastes.